builder/azure/arm/builder.go

```diff
@@ -390,7 +390,12 @@ func (b *Builder) getBlobAccount(ctx context.Context, client *AzureClient, resou
 func (b *Builder) configureStateBag(stateBag multistep.StateBag) {
 	stateBag.Put(constants.AuthorizedKey, b.config.sshAuthorizedKey)
 
-	stateBag.Put(constants.ArmTags, b.config.AzureTags)
+	tagsMap := make(map[string]*string)
+	for key, value := range b.config.AzureTags {
+		tagsMap[key] = &value
+	}
+	// Azure SDK expects string pointers for the values
+	stateBag.Put(constants.ArmTags, tagsMap)
```

Comment:
PR #9985 introduced a helper function `azcommon.MapToAzureTags` that converts tags from a simple map[string]string to a map[string]*string. 

This was not caught by the tests for step_create_resource because test for that step create the expected type. We need to add some test for this case and then update this line to look like 
```suggestion
   stateBag.Put(constants.ArmTags, azcommon.MapToAzureTags(b.config.AzureTags))
```

With that said we should probably add a type assertion check on [here](https://github.com/hashicorp/packer/blob/bdb1c47532655ceac13a14d06bc7e228b33db983/builder/azure/arm/step_create_resource_group.go#L64) to capture any failed conversions which will also prevent the Packer crash in #10012 . 

Something like

```go
tags, ok := state.Get(constants.ArmTags).(map[string]*string)
if !ok {                                        
    err := fmt.Errorf("failed to extract tags from state bag")                                                                                                                                                                                                                                      
    state.Put(constants.Error, err)                                                                                                                                                                                                                                                                    
    s.error(err)                                                                                                                                                                                                                                                                                  
    return multistep.ActionHalt                                                                                                                                                                                                                                                                        
  } 
```